modules/build/src/main/scala/scala/build/errors/Severity.scala

```diff
@@ -0,0 +1,8 @@
+package scala.build.errors
+
+sealed trait Severity
```

Comment:
NIT To define enums in Scala, we usually extends `Product with Serializable`, and have all the sub-types be case classes or case objects (so that they have a nicer to string, behave better with matches, …). And `abstract class` generates less classes I think (or only used to maybe), and explicitly says this is not really an "interface", more the root of a class hierarchy.
```suggestion
sealed abstract class Severity extends Product with Serializable
```